pkg/proxy/hybrid/proxy.go

```diff
@@ -97,11 +104,7 @@ func (p *HybridProxier) OnEndpointsUpdate(endpoints []api.Endpoints) {
 	forIPTables := []api.Endpoints{}
 
 	for _, endpoint := range endpoints {
-		svcName := types.NamespacedName{
-			Namespace: endpoint.Namespace,
-			Name:      endpoint.Name,
-		}
-		if _, ok := p.usingUserspace[svcName]; !ok {
+		if !p.isUsingUserspace(endpoint.Namespace, endpoint.Name) {
```

Comment:
I would just lock around this entire block, instead of locking on each operation like this.